### PR TITLE
Feature: Add wiki comments to v2 API

### PR DESCRIPTION
**Pull request by samanehsan**
Fixes JIRA ticket OSF-5240 -- Adds ability to comment on wiki pages via the v2 API. A follow-up PR will include the front-end changes to enable wiki comments on the OSF.

<h3>TODO:</h3>

- [x] API: List, create, view, edit, delete, undelete, report, unreport
- [x] API Tests
- [x] Fix embedding comment targets
- [x] Logging
- [x] Notification emails

### CODE REVIEW

File: api/base/serializers.py [hunk 1]
```diff
@@ -541,11 +545,13 @@ def resolve(self, resource):
         """
         Resolves the view for target node or target comment when embedding.
         """
-        view_info = self.view_map.get(resource.target._name, None)
+        view_info = self.view_map.get(resource.target.referent._name, None)
```
Comment (mfraezz) [inline api/base/serializers.py]: Could this have unintended consequences for `TargetFields` that don't target a comment?

**Comment by mfraezz** [comment]
>Adds ability to comment on wiki pages via the v2 API.

Does this mean there are intentionally no UI changes to the wiki -- that comments can only be added/modified via APIv2?

File: api_tests/comments/views/test_comment_detail.py [hunk 2]
```diff
@@ -524,18 +465,45 @@ def test_public_node_view_only_link_user_cannot_see_deleted_comment(self):
...
+        expected_type = 'nodes'
+        assert_equal(res.status_code, 200)
+        assert_equal(urlparse(url).path, expected_url)
+        assert_equal(target_type, expected_type)
+
+    def test_public_node_non_contributor_commenter_can_update_comment(self):
+        project = ProjectFactory(is_public=True, comment_level='public')
+        comment = CommentFactory(node=project, user=self.non_contributor)
+        url = '/{}comments/{}/'.format(API_BASE, comment._id)
+        payload = {
```
Comment (mfraezz) [inline api_tests/comments/views/test_comment_detail.py]: Can you not use `self._set_up_payload` here?

File: api/base/serializers.py (hunk 1 above)
Comment (samanehsan) [inline api/base/serializers.py]: A comment's target should always be a `Guid` object -- it's the `Guid.referent` that would refer to the comment, node or file

Comment (mfraezz) [inline api/base/serializers.py]: Right. I guess my question could be better phrased as "Will `TargetField` ever be used for anything other than comments?"

Comment (samanehsan) [inline api/base/serializers.py]: I don't think so -- I wrote it specifically for comment targets

File: api_tests/comments/views/test_comment_detail.py (hunk 2 above)
Comment (samanehsan) [inline api_tests/comments/views/test_comment_detail.py]: I made this change in the new PR!

**Comment by samanehsan** [comment]
Replaced by #5298

**Pull request closed without merging**